File: app/src/main/java/com/bank/service/ui/statements/StatementsAdapter.java
package com.bank.service.ui.statements;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import com.bank.service.R;
import com.bank.service.ui.statements.domain.model.Statements;

public class StatementsAdapter extends RecyclerView.Adapter<StatementsAdapter.MyViewHolder> {

    private List<Statements> stateList;

    public class MyViewHolder extends RecyclerView.ViewHolder {

        public TextView st_title, st_desc, st_date, st_value;

        public MyViewHolder(View view) {
            super(view);

            st_title = (TextView) view.findViewById(R.id.st_title);
            st_desc =  (TextView) view.findViewById(R.id.st_desc);
            st_date =  (TextView) view.findViewById(R.id.st_date);
            st_value = (TextView) view.findViewById(R.id.st_value);
        }
    }

    public StatementsAdapter(List<Statements> stateList) {
        this.stateList = stateList;
    }

    @Override // @NonNull
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
                 View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.statements_list_row, parent, false);

        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {

        Statements stm = stateList.get(position);

         holder.st_title.setText(stm.getTitle());
         holder.st_desc.setText(stm.getDesc());
         holder.st_date.setText(stm.getDate());
         holder.st_value.setText(stm.getValue());
    }

    @Override
    public int getItemCount() {
        return stateList.size();
    }
}

File: app/src/main/java/com/joaoneto/testeandroidv2/mainScreen/view/adapter/StatementsAdapter.kt
package com.joaoneto.testeandroidv2.mainScreen.view.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.joaoneto.testeandroidv2.R
import com.joaoneto.testeandroidv2.databinding.AdapterItemListBinding
import com.joaoneto.testeandroidv2.mainScreen.model.StatementModel
import com.joaoneto.testeandroidv2.util.system.Formatter

class StatementsAdapter(private val statements: List<StatementModel>) :
    RecyclerView.Adapter<StatementsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(
            DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                R.layout.adapter_item_list,
                parent,
                false
            )
        )


    override fun getItemCount(): Int = statements.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
        holder.bindView(statements[position])

    class ViewHolder(private val binding: AdapterItemListBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bindView(statement: StatementModel) {

            binding.executePendingBindings()
            binding.statement = statement
            binding.formatter = Formatter()

        }
    }
}